rpc/backend/src/main/scala/io/udash/rpc/utils/FileDownloadServlet.scala

```diff
@@ -0,0 +1,33 @@
+package io.udash.rpc.utils
+
+import java.io.File
+import java.nio.file.Files
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+
+/** Template of a servlet handling files download. */
+abstract class FileDownloadServlet extends HttpServlet {
+  /** Resolves a file basing on the request parameters. */
+  protected def resolveFile(request: HttpServletRequest): File
+
+  /** Returned value will be send to the client as a file name.
+    * By default returns the original file name. */
+  protected def presentedFileName(name: String): String = name
+
+  override def doGet(request: HttpServletRequest, response: HttpServletResponse): Unit = {
+    val file = resolveFile(request)
+
+    if (!file.exists()) response.sendError(404, "File not found!")
+    else {
+      val context = getServletContext
+      val mimeType = Option(context.getMimeType(file.getAbsolutePath)).getOrElse("application/octet-stream")
+
+      response.setContentType(mimeType)
+      if (file.length() <= Int.MaxValue) response.setContentLength(file.length.toInt)
```

Comment:
http://stackoverflow.com/questions/11466905/how-to-set-content-length-as-long-value-in-http-header-in-java